jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -216,21 +216,27 @@ YUI.add('user-profile', function() {
         this.setState({bundleList: data});
       }
     },
-
     /**
+
       Take the supplied UUID, fetch the username and password then call the
-      passed in switchModel method.
+      passed in switchModel method. If the UUID and name are null, simply
+      disconnect without reconnecting to a new model.
 
-      @method switchEnv
-      @param {String} uuid The env UUID.
-      @param {String} name The env name.
+      @method switchModel
+      @param {String} uuid The model UUID.
+      @param {String} name The model name.
     */
-    switchEnv: function(uuid, name) {
+    switchModel: function(uuid, name) {
       var props = this.props;
       props.showConnectingMask();
-      props.switchModel(uuid, this.state.envList);
-      props.dbEnvironmentSet('name', name);
       this.close();
+      if (!uuid && !name) {
+        // No parameters = disconnect with no reconnect.
+        props.switchModel();
+      } else {
+        props.switchModel(uuid, this.state.envList);
```

Comment:
because the utils `switchModel` method requires the existence of this uuid then I feel this entire conditional can be truncated down to simply `props.switchModel(uuid, this.state.envList);` and rely on the utils `switchModel` to do the right thing.